src/search.cpp

```diff
@@ -22,6 +22,7 @@
 #include <cassert>
 #include <cmath>
 #include <cstring>   // For std::memset
+#include <deque>
```

Comment:
Minor nitpick: perhaps a queue might provide better performance?

Reply:
I briefly considered vector at the time, but didn't see std::queue ... but looking at it, it seems to be deque under the hood anyway?
Thinking about it now, I suspect a simple array with an index that wraps around would do, and perhaps perform a bit better.

Reply:
std::queue is a container adapter, uses std::deque by default. I'd say std::deque is the right choice here. You're right that a simple array would be faster, because deque doesn't work as a ring buffer and has to allocate new nodes after some iterations, but this looks like a completely irrelevant code section from performance point of view.

Reply:
I ran STC for an array version against the deque version and the array came out slightly ahead:

STC :
LLR: -2.96 (-2.94,2.94) [-1.50,4.50]
Total: 69359 W: 15220 L: 15125 D: 39014
https://tests.stockfishchess.org/tests/view/5de9f054caa7c610e4d1868f

so I would use the array instead of the deque but no strong preference either way. Thoughts?

Reply:
can you try array[8] and(index+1)&7 instead of %5?

Reply:
My guess is even numbers are best since that generally compares positions after the same side has moved.
But thanks for trying a single variable and suggesting using a bitwise operator instead of `%`. Very helpful :-)

Once the current test finishes I'll create an array version with 2/4 elements and 6,6 for the constants and test it from scratch, either against master as a replacement for this patch, or against this patch for a second PR.

Reply:
>  since that generally compares positions after the same side has moved.
Can this be simplified to Array[WHITE]/Array[BLACK] for each side_to_move instead of ^1?

Reply:
I'm not sure. Maybe we can get rid of ` iterIdx ` altogether and just use ` rootDepth & 1 ` when we read and assign the array in lines 533 and 561?